signac/contrib/project.py

```diff
@@ -1608,7 +1609,7 @@ def get_project(cls, root=None, search=True, **kwargs):
                 )
             )
 
-        return cls(config=config, **kwargs)
+        return cls(root=config["project_dir"], **kwargs)
```

Comment:
The current approach finds and loads a project config and then uses its `"project_dir"` key to construct the `Project`. Can this be simplified somehow, so that we check for config _existence_ and then pass the `root` to the constructor? We shouldn't need to load the config twice (in `get_project` and again in the constructor), right?

Reply:
Hmm that's a good point. This question is related to the last piece of #643, which is changing `load_config` to not populate the `project_dir` into the config but to instead change `load_config` to return a tuple of `(root_dir, config)`. The reason I originally wanted to make that change is because actually modifying the contents of the config with something that isn't contained in any of the config files is an unexpected side effect that could cause confusion. Perhaps your question here is an indication that the discovery aspect should be separated into a completely separate function, and that `load_config` would stop discovering altogether? Alternatively, `load_config` could still be allowed to discover even if we added a separate `discover_config` function, but that seems like unnecessary redundancy.

Reply:
@bdice let me know what you think here. I can split those changes into a separate PR if that's the direction that we want to go.

Reply:
It might be a better design to separate discovery from config loading (parsing). A separate PR might be best, I agree.